app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -102,6 +102,8 @@ class MessagingManager @Inject constructor(
         const val TAG = "MessagingService"
 
         const val APP_PREFIX = "app://"
+        const val DEEP_LINK_PREFIX = "deep-link://"
+        const val INTENT_PREFIX = "intent://"
```

Comment:
This is incorrect (or at least, it is used incorrectly): only the `intent:` part is always part of a intent URI, `//` is not required. 

Example: the output of the following Intent `toUri` from the debug app:
```kotlin
Intent(context, NfcSetupActivity::class.java).toUri(Intent.URI_INTENT_SCHEME)
```
is:
```
intent:#Intent;component=io.homeassistant.companion.android.debug/io.homeassistant.companion.android.nfc.NfcSetupActivity;end
```

Reply:
This is the same format that we use for the WebView activity.

We can even see the same format used in the anchor link example in the chrome docs.

Reply:
The Chrome docs also show that only `intent:` is always included, the `//` in their example is from the next line "HOST/URI-path" which is marked optional

Reply:
Well I would hate to introduce a breaking change and/or have 2 different methods used in each place. The code still works FWIW.

Reply:
But it doesn't currently work? Taking the (valid) intent URI from my previous comment I'm sending a notification to the app on commit 29f61e770060df255d42a30d310e5890628d9d78:

```yaml
service: notify.mobile_app_pixeldev
data:
  message: test
  data:
    actions:
      - action: URI
        title: test
        uri: >-
          intent:#Intent;component=io.homeassistant.companion.android.debug/io.homeassistant.companion.android.nfc.NfcSetupActivity;end
```

Tapping this action doesn't open the NFC activity as expected but instead it opens the webview (`else` branch, and it tries to append it to my instance's URL? `http://instance/intent:?external_auth=1#Intent;component=io.homeassistant.companion.android.debug/io.homeassistant.companion.android.nfc.NfcSetupActivity;end`). If I change `INTENT_PREFIX` to only `intent:`, it works as expected.

> Well I would hate to introduce a breaking change

Intent URI support is new in this PR, so no breaking change yet?

Reply:
> But it doesn't currently work?

Try the URI example I used up above in the PR description, it does indeed launch the scanner app to scan something.


> > Well I would hate to introduce a breaking change
> 
> Intent URI support is new in this PR, so no breaking change yet?

Its only new for notifications but not in webview.

We currently already support `intent://` in [webview links](https://companion.home-assistant.io/docs/integrations/android-webview#links) so for consistency I would like them to remain the same so a user can use the same code in both places.

Reply:
Guess I found a bug, the same intent link doesn't work in the webview :)

To be clear, I don't think this would be a breaking change as it's checking for the start of the link and `intent://` and `intent:` both start with `intent:`.

Reply:
I suppose we could just drop the `//` from the prefix and it should still work for all cases.  For the example purpose I will keep it alone so a user can see a working example.  

Thanks for clearing that up 😅 I was stuck on what we did previously as the correct way since we had it for so long.

Reply:
Good catch, thanks for sticking with me while I was stuck in my old ways